File: examples/node/studio_connect_agent_router.js
// Connected MicroResolve client for the agent-tool-router demo.
//
// Subscribes to a running Studio at http://localhost:3001 and keeps a
// local copy of the `agent` namespace in ./data-node/.
//
// Usage:
//   node agent_router_client.js status
//   node agent_router_client.js watch
//   node agent_router_client.js resolve "cancel my subscription"

const { MicroResolve } = require('microresolve');

const SERVER = 'http://localhost:3001';
const NAMESPACE = 'agent';
const DATA_DIR = './data-node';
const TICK_SECS = 5;

const mr = new MicroResolve({
  serverUrl: SERVER,
  subscribe: [NAMESPACE],
  tickIntervalSecs: TICK_SECS,
  dataDir: DATA_DIR,
});
const ns = mr.namespace(NAMESPACE);

const cmd = process.argv[2] || 'status';

if (cmd === 'status') {
  console.log(`version=${ns.version()}  intents=${ns.intentCount()}`);
  process.exit(0);
}

if (cmd === 'watch') {
  console.log(`watching '${NAMESPACE}' on ${SERVER} (tick every ${TICK_SECS}s)…`);
  let last = -1;
  setInterval(() => {
    const v = ns.version();
    if (v !== last) {
      console.log(`v${v}  intents=${ns.intentCount()}`);
      last = v;
    }
  }, 1000);
}

if (cmd === 'resolve') {
  const query = process.argv.slice(3).join(' ');
  if (!query) {
    console.error('usage: node agent_router_client.js resolve <query…>');
    process.exit(2);
  }
  const matches = ns.resolve(query);
  if (matches.length) {
    console.log(`${matches[0].id}  score=${matches[0].score.toFixed(2)}`);
  } else {
    console.log('(no match)');
  }
  process.exit(0);
}

File: examples/python/studio_connect_agent_router.py
"""Connected MicroResolve client for the agent-tool-router demo.

Subscribes to a running Studio at http://localhost:3001 and keeps a
local copy of the `agent` namespace in ./data-python/.

Usage:
  python agent_router_client.py status                              # version + intent count
  python agent_router_client.py watch                               # live: print version every change
  python agent_router_client.py resolve "cancel my subscription"    # one-shot resolve
"""

import sys
import time

from microresolve import MicroResolve

SERVER = "http://localhost:3001"
NAMESPACE = "agent"
DATA_DIR = "./data-python"
TICK_SECS = 5

mr = MicroResolve(
    server_url=SERVER,
    subscribe=[NAMESPACE],
    tick_interval_secs=TICK_SECS,
    data_dir=DATA_DIR,
)
ns = mr.namespace(NAMESPACE)


def main() -> int:
    cmd = sys.argv[1] if len(sys.argv) > 1 else "status"

    if cmd == "status":
        print(f"version={ns.version()}  intents={ns.intent_count()}")
        return 0

    if cmd == "watch":
        print(f"watching {NAMESPACE!r} on {SERVER} (tick every {TICK_SECS}s)…")
        last = -1
        while True:
            v = ns.version()
            if v != last:
                print(f"v{v}  intents={ns.intent_count()}")
                last = v
            time.sleep(1)

    if cmd == "resolve":
        query = " ".join(sys.argv[2:])
        if not query:
            print("usage: agent_router_client.py resolve <query…>", file=sys.stderr)
            return 2
        matches = ns.resolve(query)
        if matches:
            print(f"{matches[0].id}  score={matches[0].score:.2f}")
        else:
            print("(no match)")
        return 0

    print(f"unknown command: {cmd}", file=sys.stderr)
    return 2


if __name__ == "__main__":
    sys.exit(main())
